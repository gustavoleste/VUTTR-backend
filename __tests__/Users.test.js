const { Users, connectDatabase } = require("../src/database/index");
const { databaseURL } = require("../src/config");
const { defaultUserOne } = require("../src/helpers/index");
require("dotenv").config();

describe("Users Model", () => {
  beforeAll(async () => {
    await connectDatabase(databaseURL, "usersModelTest");
  });

  afterAll(async () => {
    const database = await connectDatabase(databaseURL, "usersModelTest");
    await database.dropDatabase();
  });

  it("should not create a empty user", async () => {
    try {
      const newUser = new Users({});
      await newUser.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a user without the name field set", async () => {
    try {
      const userWithoutName = { ...defaultUserOne, name };
      const newUser = new Users(userWithoutName);
      await newUser.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a user without the email field set", async () => {
    try {
      const userWithoutEmail = { ...defaultUserOne, email };
      const newUser = new Users(userWithoutEmail);
      await newUser.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should not create a user without the password field set", async () => {
    try {
      const userWithoutPassword = { ...defaultUserOne, password };
      const newUser = new Users(userWithoutPassword);
      await newUser.save();
    } catch (error) {
      expect(error).toMatchSnapshot();
    }
  });

  it("should create a new user", async () => {
    try {
      const newUser = new Users(defaultUserOne);
      await newUser.save();
      expect(newUser).toMatchSnapshot();
    } catch (error) {
      console.log(error);
    }
  });
});
