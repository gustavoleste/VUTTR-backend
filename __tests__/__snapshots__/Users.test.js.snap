// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Model should create a new user 1`] = `
Object {
  "__v": 0,
  "_id": "1ceda7b37085d444ec1bec11",
  "createdAt": 2019-05-28T21:28:37.290Z,
  "email": "userone@email.com",
  "name": "user one",
  "password": "abc123",
  "role": "user",
  "updatedAt": 2019-05-28T21:28:37.290Z,
}
`;

exports[`Users Model should not create a empty user 1`] = `[ValidationError: Users validation failed: password: Path \`password\` is required., email: Path \`email\` is required., name: Path \`name\` is required.]`;

exports[`Users Model should not create a user without the email field set 1`] = `[ReferenceError: email is not defined]`;

exports[`Users Model should not create a user without the name field set 1`] = `[ReferenceError: name is not defined]`;

exports[`Users Model should not create a user without the password field set 1`] = `[ReferenceError: password is not defined]`;
